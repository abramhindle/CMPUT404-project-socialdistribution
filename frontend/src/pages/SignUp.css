:root {
    --title-size: 25px;
}

.title{
    margin-top: 30px;
    margin-left: 30%;
    font-size: var(--title-size);
}

.text{
    font-size: 20px; 
    margin-left: 300px;
}


.username_box, .email_box, .password_box, .confirm_password_box{
    border:1px solid rgb(80, 80, 80);
    border-radius:10px;
    width: 80%;
    height: 40px;
    margin-top: 10px;
    margin-left: 80px;
    margin-bottom: 20px;
    padding-left: 10px;
    min-width: 150px;
    max-width: 400px;
}


.btn{
    height: 40px;
    width: 160px;
    margin-left: 35%;
    border:none;
    color:#fff;
    background-color: #ea8c37;
    font-size: 16px;
    border-radius: 10px;
    margin-top: 10px;
    cursor: pointer;
    
}
/**wrong message notification**/
.msg{
    color: red;
    font-size: 14px;
    padding-left: 40px;
    padding-top: 10px;
    clear: both;
    font-weight: bold;
}
