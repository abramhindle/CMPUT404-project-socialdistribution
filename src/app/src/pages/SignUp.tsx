import TextField, { TextFieldProps } from '@mui/material/TextField';
import Button, { ButtonProps } from '@mui/material/Button';
import { styled as Styled } from '@mui/material/styles';
import { useState } from 'react';

interface User {
  email: string;
  username: string;
  password: string;
}
const validate = (props: 'email' | 'password' | 'username', item: string) => {
  if (props === 'email') {
    return String(item)
      .toLowerCase()
      .match(
        /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
      );
  }
  if (props === 'username') {
    if (item.length > 16) return false;
    return true;
  }

  if (props === 'password') {
    if (item.length > 16) return false;
    else {
      return String(item)
        .toLowerCase()
        .match(/^[a-z0-9]+$/i);
    }
  }
};

const ColorButton = Styled(Button)<ButtonProps>(({ theme }) => ({
  color: theme.palette.getContrastText('#e6c9a8'),
  padding: '10px',
  marginTop: '2%',
  backgroundColor: '#e6c9a8',
  '&:hover': {
    backgroundColor: '#D4AF85',
  },
}));

const TextFieldCustom = Styled(TextField)<TextFieldProps>(({ theme }) => ({
  color: theme.palette.getContrastText('#fbf8f0'),
  height: '50px',
  marginTop: '3%',
  '& .Mui-focused': {
    color: 'black',
  },
  backgroundColor: 'white',
  '&:hover': {
    backgroundColor: '#fbf8f0',
  },
}));

export default function SignUp() {
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [validEmail, setValidEmail] = useState(true);
  const [validPassword, setValidPassword] = useState(true);
  const [validUsername, setValidUsername] = useState(true);

  const handleClick = () => {
    const user: User = {
      username: username,
      email: email,
      password: password,
    };

    validate('email', email) ? setValidEmail(true) : setValidEmail(false);
    validate('username', username) ? setValidUsername(true) : setValidUsername(false);
    validate('password', password) ? setValidPassword(true) : setValidUsername(false);

    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
    validEmail && validPassword && validUsername
      ? localStorage.setItem('user', JSON.stringify(user))
      : null;

    //Discuss this^
  };
  const onChangeUsername = (event: any) => {
    setUsername(event?.target?.value);
  };
  const onChangeEmail = (event: any) => {
    setEmail(event?.target?.value);
  };
  const onChangePassword = (event: any) => {
    setPassword(event?.target?.value);
  };
  return (
    <>
      {validUsername ? (
        <TextFieldCustom
          id="standard-basic"
          color="primary"
          label="Username"
          variant="standard"
          onChange={onChangeUsername}
          helperText="Upto 16 characters."
        />
      ) : (
        <TextFieldCustom
          id="standard-error"
          color="primary"
          label="Username"
          variant="standard"
          onChange={onChangeUsername}
          helperText="Incorrect username"
        />
      )}
      {validEmail ? (
        <TextFieldCustom
          id="standard-basic"
          defaultValue={email}
          color="primary"
          label="Email"
          variant="standard"
          onChange={onChangeEmail}
        />
      ) : (
        <TextFieldCustom
          error
          id="outlined-error"
          defaultValue={email}
          label="Email"
          onChange={onChangeEmail}
          helperText="Incorrect email"
        />
      )}
      {validPassword ? (
        <TextFieldCustom
          id="standard-basic"
          label="Password"
          variant="standard"
          type="password"
          autoComplete="current-password"
          onChange={onChangePassword}
          helperText="Alphanumeric 16 characters."
        />
      ) : (
        <TextFieldCustom
          id="standard-error"
          label="Password"
          variant="standard"
          type="password"
          autoComplete="current-password"
          onChange={onChangePassword}
          helperText="Alphanumeric 16 characters!"
        />
      )}
      <ColorButton variant="contained" onClick={handleClick}>
        Sign up
      </ColorButton>
    </>
  );
}
