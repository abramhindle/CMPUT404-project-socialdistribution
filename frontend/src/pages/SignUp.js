import Headers from "../components/Headers";
import "./SignUp.css"

function SignUp() {
    return (
    <div className="SignUp">
        <Headers />
        <div class="signup-wrap">
            <form id="user_signup" action="">
                <ol class="title">Creat a new account</ol>
                <tr>
                    <ol class="text">username:</ol>
                    <td width="100%">
                        <input class="username_box" name="" id="accountName" type="text"/>
                    </td>
                </tr>

                <tr>
                    <ol class="text">email:</ol>
                    <td width="100%">
                        <input class="email_box" name="email" id="email" type="text"/>
                    </td>
                </tr>


                <tr>
                    <ol class="text">password:</ol>
                    <td width="100%">
                        <input class="password_box" name="password" id="password" type="text"/>
                    </td>
                </tr>

                <tr>
                    <ol class="text">confirm_password:</ol>
                    <td width="100%">
                        <input class="confirm_password_box" name="confirm_password" id="confirm_password" type="text"/>
                    </td>
                </tr>
        
                <button class="btn" onclick="return check(this.form);">Create account</button>


                <div id="CheckMsg" class="msg"></div>
            </form>
        </div>

    </div>
  
    );
}
export default SignUp;

  
function check(form){
    return true;
}
function register(form){
    return true;
}