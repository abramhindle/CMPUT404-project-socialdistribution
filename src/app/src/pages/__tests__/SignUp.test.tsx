import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';

import SignUp from '../SignUp';

let container: any = null;
//issues with container type and its methods

beforeEach(() => {
  // setup a DOM element as a render target
  container = document?.createElement('div');
  document?.body?.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container?.remove();
  container = null;
});

describe('SignUp suite', () => {
  it('Renders correctly', () => {
    const tree = renderer.create(<SignUp setCurrentUser={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
