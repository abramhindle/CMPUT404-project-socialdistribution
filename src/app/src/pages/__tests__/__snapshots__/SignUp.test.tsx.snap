// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp suite Renders correctly 1`] = `
Array [
  <div
    className="MuiFormControl-root MuiTextField-root css-1xkbzzz-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Username
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-describedby="standard-basic-helper-text"
        aria-invalid={false}
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
      id="standard-basic-helper-text"
    >
      Upto 16 characters.
    </p>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root css-1xkbzzz-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Email
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        defaultValue=""
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>,
  <div
    className="MuiFormControl-root MuiTextField-root css-1xkbzzz-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="standard-basic"
      id="standard-basic-label"
    >
      Password
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-describedby="standard-basic-helper-text"
        aria-invalid={false}
        autoComplete="current-password"
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={false}
        id="standard-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
      id="standard-basic-helper-text"
    >
      Alphanumeric 16 characters.
    </p>
  </div>,
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1ohtnfp-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Sign up
  </button>,
]
`;
